builtin/providers/github/resource_github_repository_deploy_key.go

```diff
@@ -0,0 +1,120 @@
+package github
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/google/go-github/github"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceGithubRepositoryDeployKey() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGithubRepositoryDeployKeyCreate,
+		Read:   resourceGithubRepositoryDeployKeyRead,
+		// Deploy keys are defined immutable in the API. Updating results in force new.
+		Delete: resourceGithubRepositoryDeployKeyDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"key": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"read_only": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+				Default:  true,
+			},
+			"repository": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"title": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceGithubRepositoryDeployKeyCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Organization).client
+
+	repo := d.Get("repository").(string)
+
+	k := d.Get("key").(string)
+	t := d.Get("title").(string)
+	key := &github.Key{
+		Key:   &k,
+		Title: &t,
+	}
+
+	if readOnly, ok := d.GetOk("read_only"); ok {
```

Comment:
We have a default value - therefore, there is no need for the `d.GetOk("")` here - it will always be there

We can replace this and do it inline in the `&github.Key` like Key and Title